group-coordinator/src/main/java/org/apache/kafka/coordinator/group/modern/consumer/ConsumerGroup.java

```diff
@@ -334,6 +342,13 @@ private void removeStaticMember(ConsumerGroupMember oldMember) {
         }
     }
 
+    /**
+     * @return The number of members subscribed to the provided regex.
+     */
+    public int numSubscribedMembers(String regex) {
+        return subscribedRegularExpressions.getOrDefault(regex, 0);
+    }
```

Comment:
from PR description

> This will be useful to know whether a regex is new in the group or whether a regex must be removed from the group.

reducing the list of subscribed regex is helpful when performing pattern matching for updated topic metadata, is this correct? 

are there any other benefits this brings?

Reply:
That's the main gain as I see it, and it's a critical one, but we can also leverage this counter in the future when defining the strategies to re-eval all regular expressions (ie. we could decide to re-eval the "most used" first based on this subscribers count)

Reply:
@jeffkbkim Yeah, the goal is to update the regex once for the entire group if it is used by multiple members. This data structure helps to de-duplicate for sure.

Reply:
makes sense. thanks!